tcp.go

```diff
@@ -20,6 +21,29 @@ var DefaultConnectTimeout = 5 * time.Second
 
 var log = logging.Logger("tcp-tpt")
 
+// try to set linger on the connection, if possible.
+func tryLinger(conn net.Conn, sec int) {
+	if lingerConn, ok := conn.(interface {
+		SetLinger(int) error
+	}); ok {
```

Comment:
this is a handful, maybe worth naming the linger interface.

Reply:
Bah!